db/postgres/postgres_types.go

```diff
@@ -155,3 +158,25 @@ type Announcement struct {
 	Message    string
 	Importance string
 }
+
+type Tag struct {
+	SMRModel
+
+	Name string `gorm:"primary_key;type:varchar(20)"`
+
+	Mods []Mod `gorm:"many2many:mod_tags"`
+}
+
+type ModTag struct {
+	SMRModel
+
+	TagID string `gorm:"primary_key;type:varchar(20)"`
+	ModID string `gorm:"primary_key;type:varchar(16)"`
+}
+
+type GuideTag struct {
+	SMRModel
```

Comment:
You don't need `SMRModel`, but rather `SMRDates` as `SMRModel` includes an `id` primary key field